From 0ccab2eecbea93087bf72b0f445358a3824bfc89 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Barnab=C3=A1s=20P=C5=91cze?=
 <barnabas.pocze@ideasonboard.com>
Date: Fri, 31 Jul 2026 14:14:10 +0200
Subject: [PATCH 31/68] ipa: libipa: agc: Omit `setLimits()` in `configure()`
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

`AgcMeanLuminance::setLimits()` is only required to be called before
`calculateNewEv()`, so the call in `configure()` has no real effect
because no calculation is carried out. So remove it.

Signed-off-by: Barnabás Pőcze <barnabas.pocze@ideasonboard.com>
Reviewed-by: Jacopo Mondi <jacopo.mondi@ideasonboard.com>
Reviewed-by: Stefan Klug <stefan.klug@ideasonboard.com>
(cherry picked from commit d51189855925915727dbcdafdf3bc219585495f1)
---
 src/ipa/libipa/agc.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/ipa/libipa/agc.cpp b/src/ipa/libipa/agc.cpp
index 93ed69896..f38c05446 100644
--- a/src/ipa/libipa/agc.cpp
+++ b/src/ipa/libipa/agc.cpp
@@ -405,9 +405,6 @@ int AgcAlgorithm::configure(agc::Session &session, agc::ActiveState &state,
 	session.maxFrameDuration = std::chrono::microseconds(frameDurations[1]);
 
 	impl_.configure(session.lineDuration, config.sensor);
-	impl_.setLimits(session.minExposureTime, session.maxExposureTime,
-			session.minAnalogueGain, session.maxAnalogueGain,
-			{});
 	impl_.resetFrameCount();
 
 	/* Configure the default exposure and gain. */
